### PR TITLE
Add python wrappers for manifolds and extend tria wrappers

### PR DESCRIPTION
**Pull request by agrayver**
This PR
- Adds possibility to work with manifolds from python. Currently SphericalManifold, PolarManifold and CylindricalManifold are supported. 
- In addition to the existing interface for the GridOut::write, adds GridIn::read support.
- Adds hyper_shell generator.

Here is an example:
image

Part of #9015

### CODE REVIEW

**Comment by agrayver** [comment]
Of course other things are easy to achieve now as well:

image

File: contrib/python-bindings/include/triangulation_wrapper.h [hunk 1]
```diff
@@ -301,6 +315,12 @@ namespace python
     void
     flatten_triangulation(TriangulationWrapper &tria_out);
 
+    /**
+     *
```
Comment (Rombur) [inline contrib/python-bindings/include/triangulation_wrapper.h]: You forgot to add the comment

**Comment by agrayver** [comment]
I added python tests.
 
P.S. Adding sensible tests required to implement more methods (reset_manifold / set_all_manifold_ids), which make this PR more complete.

File: contrib/python-bindings/source/triangulation_wrapper.cc [hunk 2]
```diff
@@ -1266,6 +1351,64 @@ namespace python
   }
 
 
+
+  void
+  TriangulationWrapper::set_manifold(const int        number,
+                                     ManifoldWrapper &manifold)
+  {
+    if ((dim == 2) && (spacedim == 2))
```
Comment (Rombur) [inline contrib/python-bindings/source/triangulation_wrapper.cc]: Can you add a check that `manifold` and `triangulation` have the `dim` and `spacedim`

File: contrib/python-bindings/tests/mapping_wrapper.py [hunk 3]
```diff
@@ -13,12 +13,6 @@
 #
 # ---------------------------------------------------------------------
 
-import os
-import sys
-module_path = os.path.abspath('/home/agrayver/lib/dealii/build_gofem/lib/python3.7/site-packages')
-if module_path not in sys.path:
-    sys.path.append(module_path)
-
```
Comment (Rombur) [inline contrib/python-bindings/tests/mapping_wrapper.py]: oops :wink:

**Comment by agrayver** [comment]
@Rombur addressed your points.

**Pull request merged**